### PR TITLE
[SDSAET-44] Introduce caches with timeouts instead of maps to prevent…

### PR DESCRIPTION
**Pull request by dominikbrandon**
https://atl.knik.put.poznan.pl/jira/browse/AET-44

### CODE REVIEW

File: core/worker/src/main/java/com/cognifide/aet/worker/impl/GrouperDispatcherFactory.java [hunk 1]
```diff
@@ -44,31 +46,52 @@
...
-      new ConcurrentHashMap<>();
+  private final Cache<SuiteTestIdentifier, AtomicInteger> lifespanCountdowns;
+  private final Cache<SuiteTestIdentifier, GrouperDispatcher> dispatchers;
+
+  public GrouperDispatcherFactory() {
+    this(CacheBuilder.newBuilder());
+  }
+
+  GrouperDispatcherFactory(CacheBuilder<Object, Object> cacheBuilder) {
+    cacheBuilder.expireAfterAccess(10, TimeUnit.MINUTES); // todo configure
```
Comment (dolatapatryk1) [inline core/worker/src/main/java/com/cognifide/aet/worker/impl/GrouperDispatcherFactory.java]: Maybe before complete this todo task, set this value to bigger one? I think huge Suite can process longer than 10 minutes

Comment (dominikbrandon) [reply core/worker/src/main/java/com/cognifide/aet/worker/impl/GrouperDispatcherFactory.java]: yup but the clocks are refreshed after each access

Comment (dominikbrandon) [reply core/worker/src/main/java/com/cognifide/aet/worker/impl/GrouperDispatcherFactory.java]: anyway i can extend it to like 20 minutes, it doesnt really matter

Comment (dominikbrandon) [reply core/worker/src/main/java/com/cognifide/aet/worker/impl/GrouperDispatcherFactory.java]: done

**Pull request merged**